runtime/mrc/src/lib.rs

```diff
@@ -695,14 +694,8 @@ impl_runtime_apis! {
 			Ok(Msa::get_owner_of(&key))
 		}
 
-		fn check_delegations(delegator_msa_ids: Vec<MessageSenderId>, provider_msa_id: MessageSenderId) -> Result<BTreeMap<MessageSenderId, bool>, DispatchError> {
-			let mut map = BTreeMap::new();
-			let provider = pallet_msa::types::Provider(provider_msa_id);
-			for id in delegator_msa_ids {
-				let delegator = pallet_msa::types::Delegator(id);
-				map.insert(id, Msa::get_provider_info_of(provider, delegator).is_some());
-			}
-			Ok(map)
+		fn has_delegation(delegator: Delegator, provider: Provider) -> Result<bool, DispatchError> {
+			Ok(Msa::get_provider_info_of(provider, delegator).is_some())
```

Comment:
What is the delegation was revoked ? we are not removing any delegations, so a revoked delegation would still be some

Reply:
That was my exact point that is_some just tells you it exists and still may have permissions

Glad you analyzed the right place 👏

Reply:
It should be matched and rather a permission and expired notice be returned to show the caller the actual config than booleans they can make their decision based on providerinfo

Reply:
Now I that I think about it